selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java

```diff
@@ -222,7 +227,13 @@ public Server getServerByExposedPort(String workspaceId, String exposedPort) thr
 
     ensureRunningStatus(workspace);
 
-    return workspace.getRuntime().getDevMachine().getRuntime().getServers().get(exposedPort);
+    Map<String, ? extends Machine> machines = workspace.getRuntime().getMachines();
+    for (Machine machine : machines.values()) {
+      if (machine.getServers().get("wsagent/http") != null) {
+        return machine.getServers().get(exposedPort);
```

Comment:
Please, use WsAgentMachineFinderUtil and get by port inside server object as now key in servers map is not port but reference. And ServerConfig should be used instead of Server

Reply:
This method is renamed to 
```public Server getServerFromDevMachineBySymbolicName(String workspaceId, String serverName)```
 because it is used in that way (getting server by it's name in the map)